lib/browser/browser.ts

```diff
@@ -34,6 +34,78 @@ patchClass('FileReader');
 propertyPatch();
 registerElementPatch(_global);
 
+// Treat XMLHTTPRequest as a macrotask.
+patchXHR(_global);
```

Comment:
Not entirely sure what you mean here - instead of patching XMLHttpRequest with `patchEventTargetMethods` like everything else, we could special-case it, but that wouldn't work for browsers that have a global `EventTarget` if I understand correctly (https://github.com/angular/zone.js/blob/9b3e77988a54824ee97a3277cc192ab442cdb788/lib/browser/event-target.ts).

We could also have a special case for XMLHttpRequest in `patchOnProperties` - I can try that out if it sounds good to you.